temporalcli/commands.operator_namespace.go

```diff
@@ -0,0 +1,318 @@
+		VisibilityArchivalState:          archivalState(c.VisibilityArchivalState.Value),
+		VisibilityArchivalUri:            c.VisibilityUri,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to create namespace: %w", err)
+	}
+	cctx.Printer.Println(color.GreenString("Namespace %s successfully registered.", nsName))
+	return nil
+}
+
+func (c *TemporalOperatorNamespaceDeleteCommand) run(cctx *CommandContext, args []string) error {
+	nsName := args[0]
+	yes, err := cctx.promptString(
+		color.RedString("Are you sure you want to delete namespace %s? Type namespace name to confirm:", nsName),
+		nsName,
+		c.Yes)
+	if err != nil {
+		return err
+	}
+
+	if !yes {
+		return fmt.Errorf("user denied confirmation or mistyped the namespace name")
+	}
+
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	resp, err := cl.OperatorService().DeleteNamespace(cctx, &operatorservice.DeleteNamespaceRequest{
+		Namespace: nsName,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to delete namespace: %w", err)
+	}
+
+	if cctx.JSONOutput {
+		return cctx.Printer.PrintStructured(resp, printer.StructuredOptions{})
+	}
+	cctx.Printer.Println(color.GreenString("Namespace %s has been deleted.", nsName))
+	return nil
+}
+
+func (c *TemporalOperatorNamespaceDescribeCommand) run(cctx *CommandContext, args []string) error {
+	nsID := c.NamespaceId
+	var nsName string
+	if nsID == "" && len(args) == 0 {
+		return fmt.Errorf("provide either namespace-id flag or namespace as an argument")
+	}
+	if nsID != "" && len(args) > 0 {
+		cctx.Printer.Println(color.YellowString("Both namespace-id flag and namespace are provided. Will use namespace Id to describe namespace"))
+	}
+	if nsID == "" {
+		nsName = args[0]
+	}
+
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	resp, err := cl.WorkflowService().DescribeNamespace(cctx, &workflowservice.DescribeNamespaceRequest{
+		Namespace: nsName,
+		Id:        nsID,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to describe namespace: %w", err)
+	}
+
+	if cctx.JSONOutput {
+		return cctx.Printer.PrintStructured(resp, printer.StructuredOptions{})
+	}
+	return printNamespaceDescriptions(cctx, resp)
+}
+
+func (c *TemporalOperatorNamespaceListCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+
+	// This is a listing command subject to json vs jsonl rules
+	cctx.Printer.StartList()
+	defer cctx.Printer.EndList()
+
+	var nextPageToken []byte
+	for {
+		resp, err := cl.WorkflowService().ListNamespaces(cctx, &workflowservice.ListNamespacesRequest{
+			NextPageToken: nextPageToken,
+			PageSize:      100,
+		})
+		if err != nil {
+			return fmt.Errorf("failed listing namespaces: %w", err)
+		}
+
+		for _, ns := range resp.GetNamespaces() {
+			if cctx.JSONOutput {
```

Comment:
Might as well put this condition around the loop instead of inside, but doesn't matter much

Reply:
If it is printed one by one as a card, it is not putting a new line between cards. Is there any option for getting a new line after each card is printed? Or do you suggest printing a new line after each card here?

Reply:
Let me merge this and try to fix that in a different PR.

Reply:
It is already around the inner loop. If I put it outside the outer loop, then we will have to create a list of all namespaces in the cluster.

Reply:
To clarify, change:

```go
		for _, ns := range resp.GetNamespaces() {
			if cctx.JSONOutput {
				_ = cctx.Printer.PrintStructured(ns, printer.StructuredOptions{})
			}
		}
```
to:
```go
		if cctx.JSONOutput {
			for _, ns := range resp.GetNamespaces() {
				_ = cctx.Printer.PrintStructured(ns, printer.StructuredOptions{})
			}
		}
```

No reason to run a loop you'll never need